Additional update instructions for .NET 3.0 (#596)
MicrosoftDocs/quantum-docs-pr: articles/install-guide/update.md

This article assumes that you already have the QDK installed. If you are installing for the first time, then please refer to the [installation guide](xref:microsoft.quantum.install).

## Updating Q# Projects 

1. First, install the latest version of the [.NET Core SDK 3.0](https://dotnet.microsoft.com/download) and run the following command in the command prompt:
```bash
dotnet --version
```
 Verify the output is 3.0.100 or higher, then follow the instructions below depending on your setup.

### In Visual Studio
 
 1. Update to the latest version of Visual Studio 2019, see [here](https://docs.microsoft.com/visualstudio/install/update-visual-studio?view=vs-2019) for instructions
 2. Open your solution in Visual Studio
 3. From the menu, select Build > Clean Solution 

>  4. Update the target framework in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)

Reviewer: ```suggestion
 4. [Update the target framework](https://docs.microsoft.com/en-us/visualstudio/ide/visual-studio-multi-targeting-overview?view=vs-2019#change-the-target-framework) in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)
```

 5. Save and close all files in your solution
 6. Select Tools > Command Line > Developer Command Prompt
 7. For each project in the solution, run the following command:
 ```bash

>  dotnet add [path\to\project_name].csproj package Microsoft.Quantum.Development.Kit`

Reviewer: ```suggestion
 dotnet add [path\to\project_name].csproj package Microsoft.Quantum.Development.Kit
```

 ```
If your projects use any other Microsoft.Quantum packages, run the command for these too. 
 8. Close the command prompt and select Build > Build Solution (do *not* select Rebuild Solution, as rebuilding will initially fail)

### In Visual Studio Code

1. In Visual Studio Code, open the folder containing the project to update
1. Select Terminal > New Terminal
1. Follow the instructions for updating using the command line

### Using the command line

1. Navigate to the folder containing your project file
2. Run the following command:
```bash
dotnet clean [project_name].csproj
```

> 3. Update the target framework in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)

Reviewer: ```suggestion
3. [Update the target framework](https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-target-frameworks) in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)
```

> The update steps depend on your development environment. Choose your environment from the following sections.

Reviewer 2: This sentence can just be deleted. 
```suggestion
```

## Updating Q# Projects 

1. First, install the latest version of the [.NET Core SDK 3.0](https://dotnet.microsoft.com/download) and run the following command in the command prompt:
```bash
dotnet --version
```

>  Verify the output is 3.0.100 or higher, then follow the instructions below for your setup.

Reviewer 2: ```suggestion
 Verify the output is 3.0.100 or higher, then follow the instructions below depending on your setup.
```

### In Visual Studio
 

>  1. Run Build > Clean Solution 

Reviewer 2: ```suggestion
 1. From the menue, select Build > Clean Solution 
```

>  1. Run Build > Clean Solution 

Reviewer 2: I think we might need to say to actually open the solution - that is not clear, especially since VS also has an open folder mode.

>  1. Update the target framework in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)

Reviewer 2: I think we might need to be more precise here?

Author: I don't think so...we're not teaching people how to edit project files, just giving the specific instructions needed for our product. I don't want to overload the update instructions, they already feel quite long.

>  1. Update the target framework in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)

Reviewer: ```suggestion
 1. [Update the target framework](https://docs.microsoft.com/en-us/visualstudio/ide/visual-studio-multi-targeting-overview?view=vs-2019#change-the-target-framework) in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)
```

 1. Save and close all files in your solution
 1. Select Tools > Command Line > Developer Command Prompt
 1. For each project in the solution, run the following command:
 ```bash

>  dotnet add Path\To\[project_name].csproj package Microsoft.Quantum.Development.Kit`

Reviewer: ```suggestion
 dotnet add [path\to\project_name].csproj package Microsoft.Quantum.Development.Kit`
```

 ```

>  Close the command prompt and select Build > Build Solution (do *not* select Rebuild Solution, as rebuilding will initially fail)

Reviewer 2: There is something missing here:
```suggestion
If your projects use any other Microsoft.Quantum packages, run the command for these too. 
1. Close the command prompt and select Build > Build Solution (do *not* select Rebuild Solution, as rebuilding will initially fail)
```

### In Visual Studio Code

1. In Visual Studio Code, open the folder containing the project to update
1. Select Terminal > New Terminal
1. Follow the instructions for updating using the command line

### Using the command line

1. Navigate to the folder containing your project file
2. Run the following commands:
```bash
dotnet clean MyProj.csproj
```

and

```bash
dotnet add package Microsoft.Quantum.Development.Kit

> ```

Reviewer 2: ```suggestion
```
if your project uses any other Microsoft.Quantum packages, run the command for these too
```

> 3. Update the target framework in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)

Reviewer 2: Actually, we might need to move this up between dotnet clean and dotnet add package, since the add package will I think complain otherwise, and the clean will fail after!

> 3. Update the target framework in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)

Reviewer 2: ```suggestion
if your project uses any other Microsoft.Quantum packages, run the command for these too.
3. Update the target framework in each of your .csproj files to netcoreapp3.0 (or netstandard2.1 if it is a library project)
```

> 1. You can now open an existing Jupyter notebook and run it with the updated QDK.

Reviewer: It's probably worth it to ask them to call `%workspace reload` from conda, to make sure caches are cleared up.

Author: we aren't shipping conda this release though, no?

Reviewer: I meant Jupyter. Doh.
